inf-clojure.el

```diff
@@ -391,12 +384,20 @@ Fallback to `default-directory.' if not within a project."
           ((file-exists-p "build.boot") "boot")
           (t nil))))
 
+(defun inf-clojure-flavor ()
+  "Determine the flavor of the project."
+  (pcase inf-clojure-repl-flavor
+    (lumo (inf-clojure--lumo-repl-command))
+    (_ nil)))
+
 (defun inf-clojure-cmd (project-type)
   "Determine the command `inf-clojure' needs to invoke for the PROJECT-TYPE."
-  (pcase project-type
-    ("lein" inf-clojure-lein-cmd)
-    ("boot" inf-clojure-boot-cmd)
-    (_ inf-clojure-generic-cmd)))
+  (if-let ((flavor-command (inf-clojure-flavor)))
```

Comment:
@bbatsov this is what I meant basically, if `inf-clojure-repl-flavor` is present, it takes precedence. It is either this is coming up with a project marker of our own, which I don't like that much.